### PR TITLE
Feature/okf context layer

### PR DESCRIPTION
**Pull request by AssafMiron**

### CODE REVIEW

File: node-backend/lib/providers/google.mjs [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    if (configuredModel && MODEL_NAME_TO_ID[configuredModel.toLowerCase()]) {
+      configuredModel = MODEL_NAME_TO_ID[configuredModel.toLowerCase()];
+    }
+
+    const input = this.buildCliInput(request);
+    const command = await this.resolveCommand();
+    const isAgy = command.includes('agy');
+    const cliDisplayName = isAgy ? 'Google Antigravity CLI (agy)' : 'Gemini CLI';
+
+    const args = ['--prompt', input, '--output-format', 'text', '--dangerously-skip-permissions'];
```
Comment (aviatam) [inline node-backend/lib/providers/google.mjs]: This passes the entire constructed prompt/context as a command-line argument. The old Gemini provider intentionally used stdin (`--prompt -`) because productOS prompts can include full project context; on Windows this can hit the ~32k command-line limit, and on Unix it can hit ARG_MAX for larger projects. It also exposes the full prompt in process listings. Can we keep the prompt payload on stdin (or otherwise stream it) and only put flags/model selection in argv?

File: node-backend/lib/ai.mjs [hunk 2]
```diff
@@ -56,9 +65,22 @@ export class AIService {
...
         case 'geminiCli':
-        case 'gemini_cli':
-          return new GeminiCliProvider(await withDetectedCommand(mergeConfig(getCfg('geminiCli', 'gemini_cli')), 'gemini'), secrets, projectPath);
+        case 'gemini_cli': {
+          const cfg = mergeConfig(getCfg('geminiCli', 'gemini_cli'));
+          const commandsToTry = (cfg.command && path.isAbsolute(cfg.command))
+            ? [cfg.command]
+            : (cfg.command && !['gemini', 'geminiCli', 'default'].includes(cfg.command))
+              ? [cfg.command, 'agy', 'gemini']
+              : ['agy', 'gemini'];
```
Comment (aviatam) [inline node-backend/lib/ai.mjs:81]: This new default preference for `agy` makes the existing provider-factory test fail on machines where a real `agy` is installed: the test puts a fake `gemini` first on PATH, but `createProvider('geminiCli', { geminiCli: {} })` now resolves the external `agy` from the enhanced Windows search paths instead. If preferring Antigravity is intentional, the test should be updated to isolate/fake `agy` too; otherwise this is a regression for explicit `geminiCli` resolution.

**Review by aviatam — COMMENTED** [review]
Reviewed the PR and found a few actionable provider/runtime issues. One additional consistency issue that I couldn't attach inline because the line wasn't changed in this PR: `getAvailableProviders()` still uses `checkCli('gemini')`, while the rest of the new Google/Antigravity flow uses `resolveCliCommand('agy', 'gemini')`. On machines with only `agy` installed, `/api/settings/providers` will omit `geminiCli` even though chat/detect/auth can support it.

File: node-backend/lib/providers/google.mjs (hunk 1 above)
Comment (AssafMiron) [reply node-backend/lib/providers/google.mjs]: Fixed in commit 38c55c16! Prompt payload is now passed via stdin (`--prompt -` and writing `input` to `child.stdin`), avoiding command-line argument limits and keeping prompt payload out of process listings.

File: node-backend/lib/ai.mjs (hunk 2 above)
Comment (AssafMiron) [reply node-backend/lib/ai.mjs:81]: Fixed in commit 38c55c16! Preferring Antigravity CLI (`agy`) with fallback to Gemini CLI (`gemini`) is intentional. We updated `parity.test.mjs` to create both `agy` and `gemini` mock executables in the test `binDir`, properly isolating the test from any globally installed system CLI.

File: node-backend/lib/project-settings.mjs [hunk 3]
```diff
@@ -57,5 +60,14 @@ export async function saveProjectSettings(projectId, rawSettings) {
     }
   }
 
+  // Materialize OKF _context directory
+  try {
+    await generateContextDirectory(projectId, settings, project);
```
Comment (aviatam) [inline node-backend/lib/project-settings.mjs]: `project` was fetched before the metadata update above, so regenerating `_context` with this stale object keeps the old `project.goal` (and old name if this code path updates it directly) in `.metadata/_context/project/overview.md`. I verified this with `createProject('Ctx Goal Test', 'old goal')` followed by `saveProjectSettings(..., { goal: 'new goal' })`; the generated overview still contained `old goal`. Can we either update the in-memory `project` after writing `metadata`, pass the updated metadata object into `generateContextDirectory`, or re-fetch the project before generating context?

**Review by aviatam — COMMENTED** [review]
Second pass after commit 38c55c1: the earlier provider comments look addressed and the full local test suite passes (`npm ci; npm test`, 122/122). I found one remaining context-generation edge case around saving changed project goals.

**Review by aviatam — APPROVED** [review]
Approved after the follow-up fix in f097c28. Provider issues are resolved, the context goal regeneration bug is fixed with regression coverage, and CI is green.
